### PR TITLE
[SPARK-8740] [PROJECT INFRA] Support GitHub OAuth tokens in dev/merge_spark_pr.py

### PR DESCRIPTION
**Pull request by JoshRosen**
This commit allows `dev/merge_spark_pr.py` to use personal GitHub OAuth tokens in order to make authenticated requests. This is necessary to work around per-IP rate limiting issues.

To use a token, just set the `GITHUB_OAUTH_KEY` environment variable.  You can create a personal token at https://github.com/settings/tokens; we only require `public_repo` scope.

If the script fails due to a rate-limit issue, it now logs a useful message directing the user to the OAuth token instructions.

### CODE REVIEW

**Comment by JoshRosen** [comment]
If you want to check your current unauthenticated rate limit: https://api.github.com/rate_limit

File: dev/merge_spark_pr.py [hunk 1]
```diff
@@ -58,9 +64,16 @@
...
     try:
-        return json.load(urllib2.urlopen(url))
+        request = urllib2.Request(url)
+        if GITHUB_OAUTH_KEY:
+            request.add_header('Authorization', 'token %s' % GITHUB_OAUTH_KEY)
+        return json.load(urllib2.urlopen(request))
     except urllib2.HTTPError as e:
-        print "Unable to fetch URL, exiting: %s" % url
+        if "X-RateLimit-Remaining" in e.headers and e.headers["X-RateLimit-Remaining"] == '0':
+            print "Exceeded the GitHub rate limit; see the instructions in dev/run-tests.py to " + \
```
Comment (JoshRosen) [inline dev/merge_spark_pr.py]: Oops, should be dev/merge_spark_pr.py.

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by andrewor14** [comment]
Merging into master thanks @JoshRosen!

**Pull request closed without merging**